app/collections/components/AdminSubtitle.tsx

```diff
@@ -1,54 +1,64 @@
-import { useMutation } from "blitz"
+import { useMutation, validateZodSchema } from "blitz"
 import { Field, Form, Formik } from "formik"
 import toast from "react-hot-toast"
 import changeSubtitle from "../mutations/changeSubtitle"
+import { useFormik } from "formik"
+import { z } from "zod"
 
 const Subtitle = ({ collection, refetchFn, isAdmin }) => {
   const [changeSubtitleMutation, { isSuccess: isSubtitleSuccess }] = useMutation(changeSubtitle)
 
+  const formik = useFormik({
+    initialValues: {
+      subtitle: collection.subtitle,
+    },
+    validate: validateZodSchema(
+      z.object({
+        subtitle: z.string().min(1).max(300),
+      })
+    ),
+    onSubmit: async (values) => {
+      toast.promise(
+        changeSubtitleMutation({
+          id: collection.id,
+          subtitle: values.subtitle,
+        }),
+        {
+          loading: "Updating subtitle...",
+          success: () => {
+            refetchFn()
+            return "Updated subtitle"
+          },
+          error: "Failed to update subtitle",
+        }
+      )
+    },
+  })
+
   return (
     <>
       {(collection.subtitle === null || !collection.public || collection.upgraded) && isAdmin ? (
         <>
-          <Formik
-            initialValues={{
-              subtitle: collection.subtitle,
-            }}
-            onSubmit={() => {}}
+          <form
+            className="module my-2 bg-white pb-2 dark:border-gray-600 dark:bg-gray-900"
+            onSubmit={formik.handleSubmit}
           >
-            <Form
-              className="my-4"
-              onBlur={(values) => {
-                if (collection.subtitle != values.target.defaultValue) {
-                  toast.promise(
-                    changeSubtitleMutation({
-                      id: collection.id,
-                      subtitle: values.target.defaultValue,
-                    }),
-                    {
-                      loading: "Updating subtitle...",
-                      success: () => {
-                        refetchFn()
-                        return "Updated subtitle"
-                      },
-                      error: "Failed to update subtitle",
-                    }
-                  )
-                }
-              }}
+            <label htmlFor="subtitle" className="sr-only">
+              subtitle
+            </label>
+            <textarea
+              rows={1}
+              id="subtitle"
+              className="w-full border-0 text-center text-sm font-medium leading-5 focus:ring-0 dark:bg-gray-900 md:text-base "
+              {...formik.getFieldProps("subtitle")}
+            />
```

Comment:
Just a thought—I wonder if it's useful to show the error here for the zod validation. Would something like this work? (ideally we want to show different errors for the lacking (0) vs too long (300+) titles, but it may be too much overhead for now?)

```suggestion
            />
            {formik.errors.subtitle && (
              <div className="border-none text-center text-sm font-medium text-gray-700 dark:text-gray-100">
                {"The subtitle should be 1 to 300 characters long."}
              </div>
            )}
```